plugins/modules/xen_orchestra_instance.py

```diff
@@ -0,0 +1,326 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2024 Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+module: xen_orchestra_instance
+short_description: Management of instances on Xen Orchestra
+description:
+  - Allows you to create/delete/restart/stop instances on Xen Orchestra.
+version_added: 10.3.0
+extends_documentation_fragment:
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: none
+  diff_mode:
+    support: none
+options:
+  api_host:
+    description: API host to XOA API.
+    required: true
+    type: str
+  user:
+    description: Xen Orchestra user.
+    required: true
+    type: str
+  password:
+    description: Xen Orchestra password.
+    required: true
+    type: str
+  validate_certs:
+    description: Verify TLS certificate if using HTTPS.
+    type: bool
+    default: true
+  use_tls:
+    description: Use wss when connecting to the Xen Orchestra API.
+    type: bool
+    default: true
+  state:
+    description: State in which the Virtual Machine should be.
+    type: str
+    choices: ['present', 'started', 'absent', 'stopped', 'restarted']
+    default: present
+  vm_uid:
+    description:
+      - UID of the target Virtual Machine. Required when O(state=absent), O(state=started), O(state=stopped) or
+        O(state=restarted).
+    type: str
+  label:
+    description: Label of the Virtual Machine to create, can be used when O(state=present).
+    type: str
+  description:
+    description: Description of the Virtual Machine to create, can be used when O(state=present).
+    type: str
+  template:
+    description:
+      - UID of a template to create Virtual Machine from.
+      - Muse be provided when O(state=present).
+    type: str
+  boot_after_create:
+    description: Boot Virtual Machine after creation, can be used when O(state=present).
+    type: bool
+    default: false
+requirements:
+  - websocket-client >= 1.0.0
+author:
+  - Samori Gorse (@shinuza) <samorigorse@gmail.com>
+'''
```

Comment:
It would be good add a reference to the Xen Orchestra API. I suggest using `seealso`, but `notes` is also acceptable.